addons/account/models/account_account.py

```diff
@@ -432,41 +628,68 @@ def code_is_available(new_code):
 
         raise UserError(_('Cannot generate an unused account code.'))
 
+    @api.depends_context('allowed_company_ids')
```

Comment:
If anything, but not really related to this task anyway
```suggestion
    @api.depends_context('company')
```